diff --git a/node_modules/three/build/three.module.js b/node_modules/three/build/three.module.js
index 8eda950..dbbc07e 100644
--- a/node_modules/three/build/three.module.js
+++ b/node_modules/three/build/three.module.js
@@ -8832,6 +8832,7 @@ class Material extends EventDispatcher {
 		this.userData = {};
 
 		this.version = 0;
+		this.uniformsNeedUpdate = false;
 
 		this._alphaTest = 0;
 
@@ -11627,7 +11628,6 @@ class ShaderMaterial extends Material {
 		};
 
 		this.index0AttributeName = undefined;
-		this.uniformsNeedUpdate = false;
 
 		this.glslVersion = null;
 
@@ -25521,12 +25521,10 @@ function WebGLMaterials( renderer, properties ) {
 			uniforms.color.value.copy( material.color );
 			uniforms.opacity.value = material.opacity;
 
-		} else if ( material.isShaderMaterial ) {
-
-			material.uniformsNeedUpdate = false; // #15581
-
 		}
 
+		material.uniformsNeedUpdate = false; // #15581
+
 	}
 
 	function refreshUniformsCommon( uniforms, material ) {
@@ -27726,7 +27724,7 @@ function WebGLRenderer( parameters = {} ) {
 
 		}
 
-		if ( refreshMaterial ) {
+		if ( refreshMaterial || material.uniformsNeedUpdate ) {
 
 			p_uniforms.setValue( _gl, 'toneMappingExposure', _this.toneMappingExposure );
 
@@ -27759,13 +27757,6 @@ function WebGLRenderer( parameters = {} ) {
 
 		}
 
-		if ( material.isShaderMaterial && material.uniformsNeedUpdate === true ) {
-
-			WebGLUniforms.upload( _gl, materialProperties.uniformsList, m_uniforms, textures );
-			material.uniformsNeedUpdate = false;
-
-		}
-
 		if ( material.isSpriteMaterial ) {
 
 			p_uniforms.setValue( _gl, 'center', object.center );
